evap/staff/importers.py

```diff
@@ -417,6 +464,83 @@ def process(cls, excel_content, test_run):
                 raise
 
 
+class PersonImporter:
+    def __init__(self):
+        self.success_messages = []
+        self.warnings = defaultdict(list)
+        self.errors = []
+
+    def process_participants(self, course, test_run, file_content=None, source_course=None):
+        if file_content:
+            user_list, self.success_messages, self.warnings, self.errors = UserImporter.process(file_content, test_run)
+        elif source_course:
+            user_list = list(source_course.participants.all())
+
+        usernames = [user.username for user in user_list]
+        already_related = course.participants.filter(username__in=usernames).all()
+
+        if already_related:
+            msg = _("The following {} user(s) are already course participants in course {}:").format(len(already_related), course.name)
+            msg += create_user_list_string_for_message(already_related)
+
+        # since the user profiles are not necessarily saved to the database, they are not guaranteed to have a pk yet which
+        # makes anything relying on hashes unusable here (for a faster list difference)
+        users_to_add = [user for user in user_list if user not in already_related]
+
+        if test_run:
+            msg = _("{} participants would be added to the course {}:").format(len(users_to_add), course.name)
+            msg += create_user_list_string_for_message(users_to_add)
+        else:
+            course.participants.add(*users_to_add)
+            msg = _("{} Participants added to course {}").format(len(users_to_add), course.name)
+            msg += create_user_list_string_for_message(users_to_add)
```

Comment:
same applies to `process_contributors`. if you don't do this, lowercase `p` in `Participants`, same for contributions below.